Objects/longobject.c

```diff
@@ -4399,8 +4427,45 @@ PyLong_GetInfo(void)
 int
 _PyLong_Init(void)
 {
+    PyInterpreterState *interp = PyThreadState_GET()->interp;
     /* initialize long_info */
     if (Long_InfoType.tp_name == 0)
         PyStructSequence_InitType(&Long_InfoType, &long_info_desc);
+    interp->long_max_str_digits = Py_LongMaxStrDigits;
+    if (interp->long_max_str_digits == -1) {
+        interp->long_max_str_digits = _PY_LONG_DEFAULT_MAX_STR_DIGITS;
+    }
     return 1;
 }
+
+
+void
+_PyLongMaxStrDigits_Init(void)
+{
+    char *env;
+    const char *endptr;
+    long maxdigits;
+
+
+    if (Py_LongMaxStrDigits >= 0 ||
+        !Py_LongMaxStrDigitsFlag)
+        return;
+
+    env = Py_GETENV("PYTHONINTMAXSTRDIGITS");
+    if (env && *env != '\0') {
+        errno = 0;
+        maxdigits = strtol(env, (char **)&endptr, 10);
+        if (*endptr != '\0' || errno == ERANGE || maxdigits < INT_MIN || maxdigits > INT_MAX ||
```

Comment:
should these be LONG_MIN and LONG_MAX?

Reply:
No, because maxdigits is type `int`. Would love it if C had `strtoi`, but well best practice seems to be using the standard `strtol` and then cast when in an acceptable range.